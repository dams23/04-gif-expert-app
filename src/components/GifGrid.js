import React, { useState, useEffect } from 'react'
import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';


export const GifGrid = ({ category }) => {
    const [images, setImages] = useState([])
    useEffect(() => {
        getGifs(category)
            .then(imgs => setImages(imgs))
    }, [category]);

    return (
        <>
            <h3>{category}</h3>
            <div className='card-grid'>
                {
                    images.map(img => (
                        <GifGridItem
                            key={img.id}
                            {...img}
                        />
                    ))
                }
            </div>
        </>
    )
}
