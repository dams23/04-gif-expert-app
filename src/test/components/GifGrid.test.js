import React from 'react';
import { shallow } from 'enzyme';
import GifGrid from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs')
describe('Pruebas sobre el componente GifGrid', () => {
    const category = 'Han Solo';
    test('debe mostrar el GifGrid', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        const wrapper = shallow(<GifGrid
            category={category}
        />);
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de mostrar la data cuando se carguen los gifs con useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'
        },
        {
            id: 'A3C',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier Cosa'

        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid
            category={category}
        />)
        //expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });
});
