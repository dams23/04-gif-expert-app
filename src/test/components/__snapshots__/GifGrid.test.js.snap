// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifGrid debe mostrar el GifGrid 1`] = `
<Fragment>
  <h3>
    Han Solo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
